src/compiler/checker.ts

```diff
@@ -1,6 +1,7 @@
 /* @internal */

```

Comment:
Added `checkExternalEmitHelpers` for private name helpers. The collision check is to make sure that WeakMap isn't declared in any scope such that it would be resolved lexically from the code generated by a class using private named instance fields, right?

Reply:
That's correct.

Reply:
Okay, I've implemented this - just need a new error message since the one we use for the global Promise collision check won't work here. Something like "Duplicate identifier 'WeakMap'. Compiler reserves name 'WeakMap' in the scope of a class with private named class fields." (Do we also need to say something about how it only reserves this name when targeting below ESNext?) Any suggestions? @DanielRosenwasser @rbuckton